### PR TITLE
Feature/14 add custom params for topics creation

### PR DESCRIPTION
**Pull request by Gataniel**
https://github.com/provectus/kafka-ui/issues/14

### CODE REVIEW

File: kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamButton.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+interface Props {
+  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void,
+  btnColor: string;
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamButton.tsx:5]: let's use `className` instead.

Comment (workshur) [reply kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamButton.tsx:5]:
```diff
- <CustomParamButton btnColor="is-danger" btnIcon='fa-minus' onClick={() => onRemove(index)} />
+ <CustomParamButton 
+  className="is-danger" 
+  type='minus' 
+  onClick={() => onRemove(index)} 
+/>
```
```js
enum CustomParamButtonType {
  plus=
  minus=
}
interface Props {
  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void,
  className: string;
  type: CustomParamButtonType;
}
```

File: kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      Object.values(CUSTOM_PARAMS_OPTIONS).map((opt: TopicCustomParamOption) => (
+        <option key={opt.name} value={opt.name}>
+          {opt.name}
+        </option>
+      ))
+    }
+  </>
+);
+
+export default React.memo(CustomParamOptions);
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamOptions.tsx:20]: 🥇

File: kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  isDisabled,
+  index,
+  name,
+}) => {
+
+  const { register, unregister, errors, getValues, triggerValidation } = useFormContext();
+  const optInputName = `${index}[name]`;
+
+  React.useEffect(
+    () => { if (isFirstParam(index)) { unregister(optInputName) } },
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamSelect.tsx:23]: we need to add eslint ASAP

File: kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  );
+
+  const selectedMustBeUniq = (selected: string) => {
+    const values: any = getValues({ nest: true });
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamSelect.tsx:27]: any?

File: kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamValue.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  isDisabled,
+  index,
+  name,
+  defaultValue,
+}) => {
+
+  const { register, unregister, errors, watch, setValue } = useFormContext();
+  const selectInputName: string = `${index}[name]`;
+  const valInputName: string    = `${index}[value]`;
+  const selectedParamName: any  = watch(selectInputName, name);
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamValue.tsx:23]: any?

File: kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParams.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { omit, reject } from 'lodash';
+
+import { TopicFormCustomParams } from 'redux/interfaces';
+import CustomParamSelect from './CustomParamSelect';
+import CustomParamValue from './CustomParamValue';
+import CustomParamAction from './CustomParamAction';
+
+const DEF_INDEX = 'default';
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParams.tsx:9]: DEFAULT_INDEX

File: kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamButton.tsx (hunk 1 above)
Comment (Gataniel) [reply kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamButton.tsx:5]: good suggestion! thanks!

File: kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamSelect.tsx (hunk 3 above)
Comment (Gataniel) [reply kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamSelect.tsx:23]: let me add it in a separate task =)

File: kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamValue.tsx (hunk 5 above)
Comment (Gataniel) [reply kafka-ui-react-app/src/components/Topics/New/CustomParams/CustomParamValue.tsx:23]: I'm a typescript newbie =) Not sure how to handle this if things like `watch()`, `getValues()` return `undefined`

which is ok?
`const selectedParamName = watch(selectInputName, name);` 
`const selectedParamName: string = watch(selectInputName, name);` 
but if I do this way, I start getting typescript warnings like: Type 'uknown' is not assignable to type 'string' or Type 'unknown' cannot be used as an index type.

Please do suggest a correct option

**Pull request merged**